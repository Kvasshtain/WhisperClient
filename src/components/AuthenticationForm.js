import React from 'react'
import PropTypes from 'prop-types'

class AuthenticationForm extends React.Component {

    state = {
        userName : '',
        userPassword : ''
    }

    onSubmit = () => {
        this.props.onSubmit(this.state.userName, this.state.userPassword)
    }

    updateUserNameValue = (eventArg) => {
        this.setState({
            userName: eventArg.target.value
        })
    }

    updateUserPasswordValue = (eventArg) => {
        this.setState({
            userPassword: eventArg.target.value
        })
    }

    render() {
        return (
            <div>
                <input type="text" onChange={this.updateUserNameValue} name="userName"/>
                <input type="text" onChange={this.updateUserPasswordValue} name="userPassword"/>
                <button type="submit" onClick = {this.onSubmit}>Send message</button>
            </div>
        )
    }
}

AuthenticationForm.propTypes = {
    onSubmit : PropTypes.func.isRequired
}

export {AuthenticationForm}