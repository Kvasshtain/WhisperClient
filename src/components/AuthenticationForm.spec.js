import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json';

import {AuthenticationForm} from './AuthenticationForm'

describe('NewMessageInput', () => {
    const mockOnSubmit = jest.fn()

    it ('should render correctly', () => {
        const testedComponent = shallow(<AuthenticationForm 
            onSubmit = {mockOnSubmit}
        />)

        expect(shallowToJson(testedComponent)).toMatchSnapshot();
      })

})

describe('when typing name', () => {
    const mockOnSubmit = jest.fn()
    const newTestUserName = 'Capitan Nemo'

    const myComponent = shallow(<AuthenticationForm
        onSubmit = {mockOnSubmit}
    />)

    beforeEach(() => {
        myComponent.find('[name="userName"]').simulate('change', {
            target: {
                value : newTestUserName,
            }
        })
    })

    it('updates inputValue field in state', () => {
        expect(myComponent.state().userName).toEqual(newTestUserName);
    })

})

describe('when typing name', () => {
    const mockOnSubmit = jest.fn()
    const newTestUserPassword = 'root'

    const myComponent = shallow(<AuthenticationForm
        onSubmit = {mockOnSubmit}
    />)

    beforeEach(() => {
        myComponent.find('[name="userPassword"]').simulate('change', {
            target: {
                value : newTestUserPassword,
            }
        })
    })

    it('updates inputValue field in state', () => {
        expect(myComponent.state().userPassword).toEqual(newTestUserPassword);
    })

})

describe('when push the button', () => {
    const mockOnSubmit = jest.fn()

    const myComponent = shallow(<AuthenticationForm
        onSubmit = {mockOnSubmit}
    />)

    beforeEach(() => {
        myComponent.find('button').simulate('click', {
            preventDefault: () => { },
        })
    })

    it('calls the props.onSubmit', () => {
        expect(mockOnSubmit).toHaveBeenCalledTimes(1);
    })

})