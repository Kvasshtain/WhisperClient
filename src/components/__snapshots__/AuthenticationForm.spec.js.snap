// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMessageInput should render correctly 1`] = `
<div>
  <input
    name="userName"
    onChange={[Function]}
    type="text"
  />
  <input
    name="userPassword"
    onChange={[Function]}
    type="text"
  />
  <button
    onClick={[Function]}
    type="submit"
  >
    Send message
  </button>
</div>
`;
